kernel/host/xmicterm/host/xmicterm.cc

```diff
@@ -59,6 +85,37 @@ struct InfoPtr {
   uint64_t debug;
 };
 
+/// Class logs into files, opening new file for every sending channel id.
+class file_logger {
+public:
+  ~file_logger() {
+    for (auto f : files) {
+      f.second->close();
+      delete f.second;
+    }
+  }
+
+  void log(uint16_t vchannel, std::string msg) {
+    auto found = files.find(vchannel);
```

Comment:
operator[] inserts new entries on first access. this can simplify the code:
```
auto& filestream = files[vchannel];
if (!filestream) { filestream = new std::ofstream(...); ... }
*filestream << msg << std::endl;
```